src/state/lottery/fetchUnclaimedUserRewards.ts

```diff
@@ -0,0 +1,136 @@
+import BigNumber from 'bignumber.js'
+import { LotteryTicket, LotteryTicketClaimData } from 'config/constants/types'
+import { UserLotteryHistory, PastLotteryRound } from 'state/types'
+import { getLotteryV2Contract } from 'utils/contractHelpers'
+import makeBatchRequest from 'utils/makeBatchRequest'
+import { BIG_ZERO } from 'utils/bigNumber'
+import { processRawTicketData } from './helpers'
+
+interface RoundDataAndUserTickets {
+  roundId: string
+  userTickets: LotteryTicket[]
+  finalNumber: string
+}
+
+const lotteryContract = getLotteryV2Contract()
+
+const getCakeRewardsForTickets = async (
+  winningTickets: LotteryTicket[],
+): Promise<{ ticketsWithRewards: LotteryTicket[]; cakeTotal: BigNumber }> => {
+  const calls = winningTickets.map((winningTicket) => {
+    const { roundId, id, rewardBracket } = winningTicket
+    return lotteryContract.methods.viewRewardsForTicketId(roundId, id, rewardBracket).call
+  })
+  const cakeRewards = (await makeBatchRequest(calls)) as string[]
+  const cakeTotal = cakeRewards.reduce((a, b) => new BigNumber(a).plus(new BigNumber(b)), BIG_ZERO)
+  const ticketsWithRewards = winningTickets.map((winningTicket, index) => {
+    return { ...winningTicket, cakeReward: cakeRewards[index] }
+  })
+  // TODO: Remove log. To help in testing when verifying winning tickets.
+  console.log('winning tickets: ', ticketsWithRewards)
+  return { ticketsWithRewards, cakeTotal }
+}
+
+const getRewardBracket = (ticketNumber: string, finalNumber: string): number => {
+  // Winning numbers are evaluated right-to-left in the smart contract, so we reverse their order for validation here:
+  // i.e. '1123456' should be evaluated as '6543211'
+  const ticketNumAsArray = ticketNumber.split('').reverse()
+  const winningNumsAsArray = finalNumber.split('').reverse()
+  const matchingNumbers = []
+
+  // The number at index 6 in all tickets is 1 and will always match, so finish at index 5
+  for (let index = 0; index < winningNumsAsArray.length - 1; index++) {
+    if (ticketNumAsArray[index] === winningNumsAsArray[index]) {
+      matchingNumbers.push(ticketNumAsArray[index])
```

Comment:
Should we handle the negative case first?
Example:
```
if (ticketNumAsArray[index] !== winningNumsAsArray[index]) {
   break;
}
// do something

Reply:
Sure, why not 👍